share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
```

Comment:
Oh, my mistake.. I tested it with `echo (commandline -po) | grep ...` and didn't realize it was `echo` which was creating the spaces.